### PR TITLE
digest algorithm clarifications

### LINKED ISSUE #60: digest format compatibility - relates to PR #56
PR #56 replaced `hash-entry` with `digest` which added a text algorithm identifier. Hash entry used an integer algorithm identifier and has a dependency on coswid. 

The goal was to allow COSE digest algorithm types. 

The digest type should support a tagged digest type so that the exact semantics of which algorithm namespace and digest format can be specified.

### PR DESCRIPTION
**Pull request by thomas-fossati**
Fix #60

### CODE REVIEW

File: draft-ietf-rats-corim.md [hunk 1]
```diff
@@ -239,7 +241,14 @@ namespace under which the value must be understood.
 ### Digest {#sec-common-hash-entry}
 
 A digest represents the value of a hashing operation together with the hash
-algorithm used.
+algorithm used.  The type of the digest algorithm identifier can be either
+`int` or `text`.  When carried as an integer value, it is interpreted according
+to the "Named Information Hash Algorithm Registry" {{-named-info}} registry.
+When it is carried as `text`, there are no requirements with regards to its
+format.  In general, the `int` encoding is RECOMMENDED.  The `text` encoding
+should only to be used when the `digest` type conveys reference value
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:249]: grammar seems wrong - maybe something like "should only be used when..."?
upper case Reference Value?

File: draft-ietf-rats-corim.md [hunk 2]
```diff
@@ -239,7 +241,14 @@ namespace under which the value must be understood.
...
+measurements that must be matched verbatim with evidence that uses the same
```
Comment (nedmsmith) [inline draft-ietf-rats-corim.md:250]: upper case MUST?
upper case Evidence?

**Review by nedmsmith — COMMENTED** [review]
A few minor tweaks needed.

File: draft-ietf-rats-corim.md (hunk 1 above)
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:249]:
```suggestion
should only be used when the `digest` type conveys reference value
```

File: draft-ietf-rats-corim.md [hunk 3]
```diff
@@ -239,7 +241,14 @@ namespace under which the value must be understood.
...
 
 A digest represents the value of a hashing operation together with the hash
-algorithm used.
+algorithm used.  The type of the digest algorithm identifier can be either
+`int` or `text`.  When carried as an integer value, it is interpreted according
+to the "Named Information Hash Algorithm Registry" {{-named-info}} registry.
+When it is carried as `text`, there are no requirements with regards to its
+format.  In general, the `int` encoding is RECOMMENDED.  The `text` encoding
+should only be used when the `digest` type conveys reference value
+measurements that must be matched verbatim with evidence that uses the same
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:250]:
```suggestion
measurements that are matched verbatim with Evidence that uses the same
```

File: draft-ietf-rats-corim.md (hunk 2 above)
Comment (thomas-fossati) [reply draft-ietf-rats-corim.md]: the must is not normative in this piece of prose.  I have replaced "must be" with "are" to remove any possible ambiguity.

File: draft-ietf-rats-corim.md [hunk 4]
```diff
@@ -239,7 +241,14 @@ namespace under which the value must be understood.
 ### Digest {#sec-common-hash-entry}
 
 A digest represents the value of a hashing operation together with the hash
-algorithm used.
+algorithm used.  The type of the digest algorithm identifier can be either
+`int` or `text`.  When carried as an integer value, it is interpreted according
+to the "Named Information Hash Algorithm Registry" {{-named-info}} registry.
```
Comment (thomas-fossati) [inline draft-ietf-rats-corim.md:246]:
```suggestion
to the "Named Information Hash Algorithm Registry" {{-named-info}}.
```

**Review by nedmsmith — APPROVED** [review]

**Pull request merged**